portalnet/src/overlay/service.rs

```diff
@@ -1127,31 +1131,122 @@ where
             "Content keys handled by offer",
         );
 
-        let utp_controller = Arc::clone(&self.utp_controller);
+        let utp_processing = UtpProcessing::from(self);
         tokio::spawn(async move {
-            let data = match utp_controller.accept_inbound_stream(cid.clone()).await {
+            let data = match utp_processing
+                .utp_controller
+                .accept_inbound_stream(cid.clone())
+                .await
+            {
                 Ok(data) => data,
                 Err(err) => {
                     debug!(%err, cid.send, cid.recv, peer = ?cid.peer.client(), content_keys = ?content_keys_string, "unable to complete uTP transfer");
+                    // Spawn a fallback FINDCONTENT task for each content key
+                    // in a payload that failed to be received.
+                    //
+                    // We spawn these additional fallback FINDCONTENT tasks using
+                    // the same semaphore permit that was initially acquired for
+                    // the ACCEPT utp stream. This might cause the utp active count
+                    // to temporarily exceed the max, but that seems preferable to
+                    // requiring additional permits for each individual content key,
+                    // which could drop the fallback request entirely if there are no
+                    // more permits available.
```

Comment:
```suggestion
                    // Spawn a fallback FINDCONTENT task for each content key
                    // in a payload that failed to be received.
                    //
                    // We spawn these additional fallback FINDCONTENT tasks using
                    // the same semaphore permit that was initially acquired for
                    // the ACCEPT utp stream. 
```

There is no "max uTP count" There is a max uTP count initated by the overlay service. RPC calls are exempt from this requirement. So we can remove that extra part of the comment. As this code is functioning as intented